### PR TITLE
docs: add v8 blog

### PR DESCRIPTION
**Pull request by lostmygithubaccount**
## Description of changes

Ibis does streaming! and more

### CODE REVIEW

File: docs/posts/ibis-version-v8-release/index.qmd [hunk 1]
```diff
@@ -0,0 +1,238 @@
...
+In collaboration with Claypot AI ([recently acquired by Voltron
+Data](https://voltrondata.com/resources/voltron-data-acquires-claypot-ai)),
+we've added the [first streaming backend with Apache
+Flink](../../backends/flink.qmd). You can check out the [blog post](TODO: link)
+and [tutorial](TODO: link) to get started with this new backend.
+
+### RisingWave
+
+In collboration with [RisingWave](https://risingwave.com), we've added the
+[second streaming backend with RisingWave](../../backends/flink.qmd) # TODO:
```
Comment (lostmygithubaccount) [inline docs/posts/ibis-version-v8-release/index.qmd:78]: need to get the RisingWave backend page added and linked

File: docs/posts/ibis-version-v8-release/index.qmd [hunk 2]
```diff
@@ -0,0 +1,238 @@
...
+
+Three new backends were added in this release.
+
+### Apache Flink
+
+In collaboration with Claypot AI ([recently acquired by Voltron
+Data](https://voltrondata.com/resources/voltron-data-acquires-claypot-ai)),
+we've added the [first streaming backend with Apache
+Flink](../../backends/flink.qmd). You can check out the [blog post](TODO: link)
+and [tutorial](TODO: link) to get started with this new backend.
```
Comment (lostmygithubaccount) [inline docs/posts/ibis-version-v8-release/index.qmd:73]: need the Flink blog/tutorial merged and linked ahead of this

File: docs/posts/ibis-version-v8-release/index.qmd [hunk 3]
```diff
@@ -0,0 +1,238 @@
...
+author: "Ibis team"
+date: "2024-02-05"
+categories:
+    - release
+    - blog
+---
+
+## Overview
+
+Ibis v8 marks the first release of data streaming backends in Ibis! We're
```
Comment (gforsyth) [inline docs/posts/ibis-version-v8-release/index.qmd:12]: `v8` vs. `8.0`?
are we going for the tomato juice market or the webassembly market?

File: docs/posts/ibis-version-v8-release/index.qmd [hunk 4]
```diff
@@ -0,0 +1,238 @@
...
+excited to begin bridging the gap between batch and streaming systems, starting
+with [Apache Flink](https://flink.apache.org/) and
+[RisingWave](https://risingwave.com) backends in this release.
+
+We've also added a new batch backend with [Exasol](https://exasol.com), bringing
+the total number of backends Ibis supports to 20+.
```
Comment (gforsyth) [inline docs/posts/ibis-version-v8-release/index.qmd:18]: supernit -- if we're going to say "total number" then we should provide a number, not a lower bound.
you can 100% ignore this

File: docs/posts/ibis-version-v8-release/index.qmd [hunk 5]
```diff
@@ -0,0 +1,238 @@
...
+
+### What is data streaming?
+
+Data streaming is a method of processing data continuously as it arrives rather
+than in batches at points in time. This is useful for applications that require
+low latency and increasingly common in modern data applications.
```
Comment (gforsyth) [inline docs/posts/ibis-version-v8-release/index.qmd:24]:
```suggestion
low latency and is increasingly common in modern data applications.
```

File: docs/posts/ibis-version-v8-release/index.qmd [hunk 6]
```diff
@@ -0,0 +1,238 @@
...
+
+Apache Flink is the most popular open-source stream processing framework, with
+numerous cloud options. RisingWave is an open-source Postgres-compatible
+streaming database with a cloud offering that is gaining popularity and
+simplifies the streaming experience.
+
+Ibis now supports both and going forward can add more streaming backends to
+unify the Python user experience across batch and streaming systems.
```
Comment (gforsyth) [inline docs/posts/ibis-version-v8-release/index.qmd:32]: maybe a qualifier here -- in some respects we've already unified this

File: docs/posts/ibis-version-v8-release/index.qmd [hunk 7]
```diff
@@ -0,0 +1,238 @@
...
+
+### Unifying batch and streaming UX in Python
+
+Regardless of whether you're using a batch or streaming data platform -- and the
```
Comment (gforsyth) [inline docs/posts/ibis-version-v8-release/index.qmd:36]:
```suggestion
Whether you're using a batch or streaming data platform -- and the
```

File: docs/posts/ibis-version-v8-release/index.qmd [hunk 8]
```diff
@@ -0,0 +1,238 @@
...
+lines are continunally blurring between them -- you'll need a frontend to
```
Comment (gforsyth) [inline docs/posts/ibis-version-v8-release/index.qmd:37]:
```suggestion
lines are continually blurring between them -- you'll need a frontend to
```

File: docs/posts/ibis-version-v8-release/index.qmd [hunk 9]
```diff
@@ -0,0 +1,238 @@
...
+and understanding new data systems. Organizations benefit from this in the form
+of lower onboarding costs, easier collaboration between teams, and better
+interfaces for data systems.
+
+We saw in the recent [one billion row challenge post how even CSV reader keyword
+arguments can differ greatly between APIs](../1brc/index.qmd#setup-1). This is
+compounded by tightly coupling a dataframe API to every query engine, whether
+batch or streaming.
+
+Ibis aims to solve this dillemma by providing a standard dataframe API that can
```
Comment (gforsyth) [inline docs/posts/ibis-version-v8-release/index.qmd:51]:
```suggestion
Ibis aims to solve this dilemma by providing a standard dataframe API that can
```

File: docs/posts/ibis-version-v8-release/index.qmd [hunk 10]
```diff
@@ -0,0 +1,238 @@
...
+
+```{python}
+batches = t.to_pandas_batches(chunk_size=200)
+for df in batches:
+    print(df.shape)
+```
+
+### range
+
+The range function can be used to create a monotonic sequence of integers:
```
Comment (gforsyth) [inline docs/posts/ibis-version-v8-release/index.qmd:162]:
```suggestion
The `range()` function can be used to create a monotonic sequence of integers:
```

for formatting consistency with the others listed

**Review by gforsyth — COMMENTED** [review]
a collection of nits.

also, in the added functionality, do we want to link to the API docs for the methods being called out?

File: docs/posts/ibis-version-v8-release/index.qmd (hunk 3 above)
Comment (lostmygithubaccount) [reply docs/posts/ibis-version-v8-release/index.qmd:12]: I prefer `v8` but might be missing some nuance there

File: docs/posts/ibis-version-v8-release/index.qmd (hunk 6 above)
Comment (lostmygithubaccount) [reply docs/posts/ibis-version-v8-release/index.qmd:32]: how if we didn't have a streaming backend before?

**Comment by lostmygithubaccount** [comment]
@gforsyth I am just worried links would break; they can always search in the top right...

File: docs/posts/ibis-v8-release/index.qmd [hunk 11]
```diff
@@ -0,0 +1,269 @@
...
+Data](https://voltrondata.com/resources/voltron-data-acquires-claypot-ai)),
+we've added the [first streaming backend with Apache
+Flink](../../backends/flink.qmd). You can check out the [blog post](TODO: link)
+and [tutorial](TODO: link) to get started with this new backend.
+
+### RisingWave
+
+In collboration with [RisingWave](https://risingwave.com), we've added the
+[second streaming backend with RisingWave](../../backends/flink.qmd) # TODO:
+update. This backend is earlier in development, but we're excited to have it as
```
Comment (lostmygithubaccount) [inline docs/posts/ibis-v8-release/index.qmd:83]: do not merge w/ these TODOs

File: docs/posts/ibis-version-v8-release/index.qmd (hunk 6 above)
Comment (gforsyth) [reply docs/posts/ibis-version-v8-release/index.qmd]: maybe it's the unclear antecedent.

What is allowing the unification?  That we "now support both" or that "going forward [we] can add more to unify..."

File: docs/posts/ibis-v8-release/index.qmd [hunk 12]
```diff
@@ -0,0 +1,269 @@
...
+categories:
+    - release
+    - blog
+---
+
+## Overview
+
+Ibis v8 marks the first release of data streaming backends in Ibis! We're
+excited to begin bridging the gap between batch and streaming systems, starting
+with [Apache Flink](https://flink.apache.org/) and
```
Comment (zhenzhongxu) [inline docs/posts/ibis-v8-release/index.qmd:14]: maybe specifically call out Flink as a streaming backend and RisingWave as a streaming database backend.

File: docs/posts/ibis-v8-release/index.qmd [hunk 13]
```diff
@@ -0,0 +1,269 @@
...
+[RisingWave](https://risingwave.com) backends in this release. We've also added
+a new batch backend with [Exasol](https://exasol.com), bringing the total number
+of backends Ibis supports to 20.
+
+[Most geospatial operations are now supported in the DuckDB
+backend](#geospatial-operations-in-duckdb), making Ibis a great local option for
+geospatial analytics.
+
+### What is data streaming?
```
Comment (zhenzhongxu) [inline docs/posts/ibis-v8-release/index.qmd:23]: data streaming => stream processing

File: docs/posts/ibis-v8-release/index.qmd [hunk 14]
```diff
@@ -0,0 +1,269 @@
...
+
+Data streaming is a method of processing data continuously as it arrives rather
```
Comment (zhenzhongxu) [inline docs/posts/ibis-v8-release/index.qmd:25]: a key component of streaming is unbounded data processing

Comment (zhenzhongxu) [reply docs/posts/ibis-v8-release/index.qmd:25]: technically you can still do stream processing on historical data using watermark, so the key differentiation between streaming and batch is unbounded processing with time semantics.

**Review by zhenzhongxu — CHANGES_REQUESTED** [review]
Added a few suggestions on concepts and nit asks.

**Review by zhenzhongxu — APPROVED** [review]
LGTM!

**Review by cpcloud — APPROVED** [review]

**Pull request merged**